src/components/FormSections/PersonalInfo.jsx

```diff
@@ -37,32 +37,36 @@ function PersonalInfo(props){
     const telephoneFieldState = (nvrfStateFields.find(item => item.uuid === phoneNumberField.uuid));
     const raceFieldState = (nvrfStateFields.find(item => item.uuid === raceField.uuid));
 
-    const checkDateValues = (type) => {
+    const checkDateValues = (e, type) => {
         let month = props.fieldData.date_of_birth_month;
         let day = props.fieldData.date_of_birth_day;
-        /* removing age validation temporarily */
-        // let year = props.fieldData.date_of_birth_year;
-        // let yearStart = year.slice(0, 2);
-        // let birthdate = year + '-' + month + '-' + day;
-        // let age = Math.floor((new Date() - new Date(birthdate).getTime()) / 3.15576e+10)
+        let year = props.fieldData.date_of_birth_year;
+        let yearStart = year.slice(0, 2);
+
+        let currentDate = new Date();
+        let currentMonth = currentDate.getMonth();
+        let currentDay = currentDate.getDate();
+        let currentYear = currentDate.getFullYear();
+        let age = currentYear - year - (currentMonth <= month && currentDay < day);
 
         if (type === "all") {
           let dobValues = [
             month.length === 2,
             day.length === 2,
-            // year.length === 4,
+            year.length === 4,
 
             month <= 12,
             month >= 1,
             day <= 31,
             day >= 1,
-            // yearStart <= 20,
-            // yearStart >= 19,
-            // age <= 110,
-            // age >= 17
+            yearStart <= 20,
+            yearStart >= 19,
+            age <= 120,
+            age >= 16
           ];
 
           if (dobValues.includes(false)) {
+            e.target.setCustomValidity(' ');
```

Comment:
@WilliamYKwon Why was this added here? It is not being used as part of the onInput() or onInvalid() functions so I am not sure why the error message is being set to empty.

Reply:
From what I understand, onInvalid() doesn't check against checkDateValues() so even if a birthdate of a 13yo is entered, it displays the error but still lets you proceed to the next steps. So I added that there to disallow one from proceeding on.

If this isn't valid to do, I can look into other ways handle this

Reply:
@WilliamYKwon Oh okay got it! I just tested again and it works, so this should be good.